### PR TITLE
Fix Instruction error

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-vm/issues/517

### CODE REVIEW

**Review by Liaojinghui — DISMISSED** [review]

File: src/Neo.VM/Instruction.cs [hunk 1]
```diff
@@ -205,14 +205,21 @@ internal Instruction(ReadOnlyMemory<byte> script, int ip) : this((OpCode)script.
                     operandSize = OperandSizeTable[(int)OpCode];
                     break;
                 case 1:
+                    if (ip >= span.Length)
+                        throw new BadScriptException($"Instrucion out of bounds. InstructionPointer: {ip}");
```
Comment (ixje) [inline src/Neo.VM/Instruction.cs:209]: Typo: Instrucion -> Instruc_t_ion

Happy to see exceptions with messages 👍

Comment (Liaojinghui) [inline src/Neo.VM/Instruction.cs:209]:
```suggestion
                        throw new BadScriptException($"Instruction out of bounds. Instruction pointer: {ip}");
```

File: src/Neo.VM/Instruction.cs [hunk 2]
```diff
@@ -205,14 +205,21 @@ internal Instruction(ReadOnlyMemory<byte> script, int ip) : this((OpCode)script.
...
                     operandSize = span[ip];
                     break;
                 case 2:
+                    if (ip + 1 >= span.Length)
+                        throw new BadScriptException($"Instrucion out of bounds. InstructionPointer: {ip}");
```
Comment (Liaojinghui) [inline src/Neo.VM/Instruction.cs:214]:
```suggestion
                        throw new BadScriptException($"Instruction out of bounds. InstructionPointer: {ip}");
```

File: src/Neo.VM/Instruction.cs [hunk 3]
```diff
@@ -205,14 +205,21 @@ internal Instruction(ReadOnlyMemory<byte> script, int ip) : this((OpCode)script.
...
                     operandSize = BinaryPrimitives.ReadUInt16LittleEndian(span[ip..]);
                     break;
                 case 4:
+                    if (ip + 3 >= span.Length)
+                        throw new BadScriptException($"Instrucion out of bounds. InstructionPointer: {ip}");
```
Comment (Liaojinghui) [inline src/Neo.VM/Instruction.cs:219]:
```suggestion
                        throw new BadScriptException($"Instruction out of bounds. InstructionPointer: {ip}");
```

File: src/Neo.VM/Instruction.cs [hunk 4]
```diff
@@ -205,14 +205,21 @@ internal Instruction(ReadOnlyMemory<byte> script, int ip) : this((OpCode)script.
...
                     operandSize = BinaryPrimitives.ReadInt32LittleEndian(span[ip..]);
-                    if (operandSize < 0) throw new BadScriptException();
+                    if (operandSize < 0)
+                        throw new BadScriptException($"Instrucion out of bounds. InstructionPointer: {ip}, operandSize: {operandSize}");
```
Comment (Liaojinghui) [inline src/Neo.VM/Instruction.cs:222]:
```suggestion
                        throw new BadScriptException($"Instruction out of bounds. InstructionPointer: {ip}, operandSize: {operandSize}");
```

**Review by Liaojinghui — APPROVED** [review]

**Pull request merged**